### PR TITLE
new operator structure

### PR DESCRIPTION
**Pull request by jjmachan**

### CODE REVIEW

File: bentoctl_lambda/aws_lambda/app.py [hunk 1]
```diff
@@ -5,15 +5,12 @@
...
 bento_service = load("./")
 print("bento service", bento_service)
-# bento_service_api = bento_service.get_inference_api(api_name)
 
-this_module = sys.modules[__name__]
 # Disable /metrics endpoint since promethues is not configured for use
 # in lambda
 DeploymentContainer.api_server_config.metrics.enabled.set(False)
-setattr(this_module, api_name, Mangum(bento_service.asgi_app))
+mangum_app = Mangum(bento_service.asgi_app, api_gateway_base_path=API_GATEWAY_STAGE)
```
Comment (yubozhao) [inline bentoctl_lambda/aws_lambda/app.py:16]: we can come back to this. need more research

File: bentoctl_lambda/schemas.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+other_schema = {
+        "deployment_name": {
+            "required": True,
+            "type": "string",
+            "help_message": "Deployment Name"
+            }
+        }
+
+# merge both dicts
+PARAMETER_SCHEMA = {**OPERATOR_SCHEMA, **other_schema}
```
Comment (yubozhao) [inline bentoctl_lambda/schemas.py:33]: comment out for now, if not in use.

File: bentoctl_lambda/templates/terraform_default.tf [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+################################################################################
+
+data "aws_ecr_repository" "service" {
+  name = var.repository_name
+}
+
+data "aws_ecr_image" "service_image" {
+  repository_name = data.aws_ecr_repository.service.name
+  image_tag       = var.image_tag
+}
```
Comment (yubozhao) [inline bentoctl_lambda/templates/terraform_default.tf:62]: should this be part of input instead of the resource

File: bentoctl_lambda/utils.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    metadata["tag"] = bento.tag
+    metadata["bentoml_version"] = ".".join(bento.info.bentoml_version.split(".")[:3])
+
+    python_version_txt_path = "env/python/version.txt"
+    python_version_txt_path = os.path.join(path, python_version_txt_path)
+    with open(python_version_txt_path, "r") as f:
+        python_version = f.read()
+    metadata["python_version"] = ".".join(python_version.split(".")[:2])
+
+    return metadata
```
Comment (jjmachan) [inline bentoctl_lambda/utils.py:21]: moved to bentoctl

File: bentoctl_lambda/create_deployable.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    # Copy app.py which handles the Lambda events
+    shutil.copy(
+        os.path.join(path_to_templates, "app.py"),
+        os.path.join(deployable_path, "app.py"),
+    )
+
+
+def create_deployable(bento_path, destination_dir, overwrite_deployable):
+    """
+    The deployable is the bento along with all the modifications (if any)
```
Comment (jjmachan) [inline bentoctl_lambda/create_deployable.py:50]: use metadata that is provided by bentoctl here

File: bentoctl_lambda/registry_utils.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        repository_id = result["repositories"][0]["registryId"]
+        repository_uri = result["repositories"][0]["repositoryUri"]
+    except ecr_client.exceptions.RepositoryNotFoundException:
+        result = ecr_client.create_repository(repositoryName=repository_name)
+        repository_id = result["repository"]["registryId"]
+        repository_uri = result["repository"]["repositoryUri"]
+    return repository_id, repository_uri
+
+
+def get_default_registry_creds(bento_name, operator_spec):
```
Comment (yubozhao) [inline bentoctl_lambda/registry_utils.py:32]: change bento_name to repository name

Comment (yubozhao) [inline bentoctl_lambda/registry_utils.py:32]: change bento_name to repository name

**Pull request merged**